helm-util/README.md

```diff
@@ -1,44 +1,105 @@
 
-```
-local helm = import 'github.com/grafana/jsonnet-libs/helm-util/helm.libsonnet';
+The [`helm.template`](#fn-helmtemplate) function to converts a Helm Chart into Jsonnet objects,
+to be consumed by tools like [Grafana Tanka](https://tanka.dev).
+
+Helm Charts are required to be available on the local file system and are
+resolved relative to the file that calls `helm.template`:
+
+```jsonnet
+local helm = (import 'github.com/grafana/jsonnet-libs/helm-util/helm.libsonnet').new(std.thisFile);
 
 {
-  version:: 'v0.16.0',
-
-  values:: {
-    installCRDs: true,
-    global: {
-      podSecurityPolicy: {
-        enabled: true,
-        useAppArmor: false,
-      },
-    },
-  },
-
-  chart: helm.template(
-    name='cert-manager',
-    chart='jetstack/cert-manager',
-    conf={
-      values: $.values,
-      flags: [
-        '--version=%s' % $.version,
-      ],
+  // render the Grafana Chart, set namespace to "test"
+  grafana: helm.template('grafana', './charts/grafana', {
+    values: {
+      persistence: { enabled: true },
+      plugins: ['grafana-clock-panel'],
     },
-  ),
+    namespace: 'test',
+  }),
 }
+
 ```
+
+For more information on that see https://tanka.dev/helm
+
+### Internals
+
+The functionality of `helm-util` is based on the `helm template` command.
+Because Jsonnet does not support executing arbitrary command for [good
+reasons](https://jsonnet.org/ref/language.html#independence-from-the-environment-hermeticity),
+a different way was required.
+
+To work around this, [Tanka](https://tanka.dev) instead binds special
+functionality into Jsonnet that provides `helm template`.
+
+This however means this library and all libraries using this library are not
+compatible with `google/go-jsonnet` or `google/jsonnet`.
+
+
+## Index
+
+* [`fn new(calledFrom)`](#fn-new)
+* [`fn patchKubernetesObjects(object, patch)`](#fn-patchkubernetesobjects)
+* [`fn patchLabels(object, labels)`](#fn-patchlabels)
+* [`fn template(name, chart, conf)`](#fn-template)
+
+## Fields
+
+### fn new
+
+```ts
+new(calledFrom)
+```
+
+`new` initiates the `helm-util` library. It must be called before any `helm.template` call:
+ > ```jsonnet
+ > // std.thisFile required to correctly resolve local Helm Charts
+ > helm.new(std.thisFile)
+ > ```
+
+
+### fn patchKubernetesObjects
+
+```ts
+patchKubernetesObjects(object, patch)
+```
+
+`patchKubernetesObjects` applies `patch` to all Kubernetes objects it finds in `object`.
+
+### fn patchLabels
+
+```ts
+patchLabels(object, labels)
+```
+
+`patchLabels` finds all Kubernetes objects and adds labels to them.
+
+### fn template
+
+```ts
+template(name, chart, conf)
+```
+
+`template` expands the Helm Chart to its underlying resources and returns them in an `Object`,
+so they can be consumed and modified from within Jsonnet.
+
+This functionality requires Helmraiser support in Jsonnet (e.g. using Grafana Tanka) and also
```

Comment:
As someone that doesn't have the full context yet, does this refer to `Helmraiser` or should it be `helm-util`? I don't see any references in tanka to Helmraiser.

Reply:
Following on this, there is a couple of additional referrences to Helmraiser in `helm-util/helm.libsonnet`:

```
helm-util/helm.libsonnet:34:  defaultLabels:: { 'app.kubernetes.io/managed-by': 'Helmraiser' },
helm-util/helm.libsonnet:41:      This functionality requires Helmraiser support in Jsonnet (e.g. using Grafana Tanka) and also
```